modules/javafx.graphics/src/main/java/com/sun/javafx/font/freetype/PangoGlyphLayout.java

```diff
@@ -80,9 +84,11 @@ private boolean check(long checkValue, String message, long context, long desc,
         return true;
     }
 
-    private long str = 0L;
+    private Map<TextRun, Long> runUtf8 = new HashMap<>();
     public void layout(TextRun run, PGFont font, FontStrike strike, char[] text) {
-
+        for (char c: text) {
+            if (c == 0) c = '\f';
```

Comment:
This won't actually do anything (it just sets a local variable that immediately goes out of scope). I don't think that we want to modify the array itself, so you will probably want to make a copy of the array in the case there is a '0' character.

As for what to replace the '0' char with, maybe a space? @prrace can probably suggest something.

Reply:
Interestingly enough, I don't see a crash due to the 0 character, but I am seeing some new assertion warnings expecting `length >= 0` even in cases where there isn't a 0 character.

Reply:
I saw that assertion as well, but that can come from cases where empty TextRun instances are used (start == end). In that case, the length is 0, which pango considers silly, but it won't crash on that (the content array is never examined in that case)

Reply:
the crash with 0 is gone with the second patch indeed (1 str per Run), which might be a consequence of doing strlen calculations before we send content down to the pango lib